control/src/lib.rs

```diff
@@ -569,6 +587,9 @@ impl<T: Config> Pallet<T> {
 
 		// SBP-M2 review: I strongly suggest emitting events in extrinsics
 		// It clarifies code and its responsibility
+		
+		// vovacha: I think that keeping it here clarifies code and its responsibility.
+		// Some of the helpers are used in multiple places.
```

Comment:
actually the flow in previous extrinsics was like suggested, check, write in utility, on success emit event in extrinsic, like this you separate concerns of public facing code and internal e.g. write ops.

Reply:
@2075  I think it makes sense of handling events, not at the extrinsic level, when:
- function is triggered by more than one extrinsic
- function is a part of a pallet's public interface
- function is triggered from hooks (on_finalize, on_initialize)

And you can find many examples in Substrate, Acala, Orml, Rmrk, etc. So, it's not a rule, but a recommendation.

Some examples:
- [Balances](https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L1157)
- [Collective](https://github.com/paritytech/substrate/blob/master/frame/collective/src/lib.rs#L756)
- [Democracy](https://github.com/paritytech/substrate/blob/master/frame/democracy/src/lib.rs#L1129)
- and many more...

Reply:
yes, one way or the other, but kind of aligned